ludwig/explain/captum.py

```diff
@@ -215,6 +187,80 @@ def explain(self) -> Tuple[List[Explanation], List[float]]:
         return self.explanations, expected_values
 
 
+def get_input_tensors(
+    model: LudwigModel, input_set: pd.DataFrame, run_config: ExplanationRunConfig
+) -> List[torch.Tensor]:
+    """Convert the input data into a list of variables, one for each input feature.
+
+    # Inputs
+
+    :param model: The LudwigModel to use for encoding.
+    :param input_set: The input data to encode of shape [batch size, num input features].
+
+    # Return
+
+    :return: A list of variables, one for each input feature. Shape of each variable is [batch size, embedding size].
+    """
+    # Ignore sample_ratio from the model config, since we want to explain all the data.
+    sample_ratio_bak = model.config_obj.preprocessing.sample_ratio
+    model.config_obj.preprocessing.sample_ratio = 1.0
+
+    config = model.config_obj.to_dict()
+    training_set_metadata = copy.deepcopy(model.training_set_metadata)
+    for feature in config[INPUT_FEATURES]:
+        preprocessing = training_set_metadata[feature[NAME]][PREPROCESSING]
+        if preprocessing.get("cache_encoder_embeddings"):
+            preprocessing["cache_encoder_embeddings"] = False
+
+    # Convert raw input data into preprocessed tensor data
+    dataset, _ = preprocess_for_prediction(
```

Comment:
Do we want to use `run_config.batch_size` also for preprocessing? Or will preprocessing not require PyTorch since we set `cache_encoder_embeddings` to False?

Reply:
Correct, we don't need the separate batch size here since we don't do the embedding during preprocessing.